tests/linearizability/lease_ids.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linearizability
+
+import (
+	"sync"
+)
+
+type leaseIdProvider interface {
+	LeaseId(int) int64
+	AddLeaseId(int, int64)
+	RemoveLeaseId(int, int64)
+}
+
+func newLeaseIdProvider() leaseIdProvider {
+	return &atomicLeaseIdProvider{m: map[int]int64{}}
+}
+
+type atomicLeaseIdProvider struct {
+	sync.RWMutex
+	m map[int]int64
```

Comment:
Please add a comment something like: `clientId to LeaseId mapping`

Reply:
updated